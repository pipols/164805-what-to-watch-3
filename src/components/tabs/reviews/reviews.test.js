import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews.jsx";

const film = {
  poster: `bg-the-grand-budapest-hotel.jpg`,
  cover: `the-grand-budapest-hotel-poster.jpg`,
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  actors: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ],
  producer: `Wes Andreson`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: `8,9`,
  ratingDescription: `Very good`,
  votes: 240,
  duration: `1h 39m`,
  reviews: [
    {
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      votes: `8,9`,
      userName: `Kate Muir`,
      reviewDate: `December 24, 2016`
    }
  ]
};

it(`<Reviews /> renders correctly`, () => {
  const tree = renderer.create(<Reviews
    reviews={film.reviews}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
