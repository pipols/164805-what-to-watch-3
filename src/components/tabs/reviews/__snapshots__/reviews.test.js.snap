// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            user name
          </cite>
          <time
            className="review__date"
            dateTime="2020-03-07T12:35:45.538Z"
          >
            March 7, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.4
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Discerning travns will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            user name
          </cite>
          <time
            className="review__date"
            dateTime="2020-03-07T12:35:45.538Z"
          >
            March 7, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.4
      </div>
    </div>
  </div>
</div>
`;
